src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,230 @@
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T unlink(Node<T> x) {
+
+        final T element = x.item;
+        final Node<T> next = x.next;
+        final Node<T> previous = x.previous;
+
+        if (previous == null) {
+            firstNode = next;
+        } else {
+            previous.next = next;
+            x.previous = null;
+        }
+
+        if (next == null) {
+            lastNode = previous;
+        } else {
+            next.previous = previous;
+            x.next = null;
+        }
+
+        x.item = null;
+        size--;
+        return element;
+    }
+
+    private void checkedElementIndex(int index) {
+        if (!isElementIndex(index)) {
+            throw new IndexOutOfBoundsException(index + " out of bounds");
+        }
+    }
+
+    private void checkPositionIndex(int index) {
+        if (!isPositionIndex(index)) {
+            throw new IndexOutOfBoundsException(index);
+        }
+    }
+
+    private boolean isElementIndex(int index) {
+        return index >= 0 && index < size;
+    }
+
+    private boolean isPositionIndex(int index) {
+        return index >= 0 && index <= size;
+    }
+
+    private Node<T> node(int index) {
+        Node<T> x;
+        if (index < (size / 2)) {
+            x = firstNode;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+        } else {
+            x = lastNode;
+            for (int i = size - 1; i > index; i--) {
+                x = x.previous;
+            }
+        }
+        return x;
+    }
+
+    private void linkedFirst(T e) {
+        final Node<T> f = firstNode;
+        final Node<T> newNode = new Node<>(null, e, f);
+        firstNode = newNode;
+        if (f == null) {
+            lastNode = newNode;
+        } else {
+            f.previous = newNode;
+        }
+        size++;
+    }
+
+    private void linkedLast(T e) {
+        final Node<T> l = lastNode;
+        final Node<T> newNode = new Node<>(l, e, null);
+        lastNode = newNode;
+        if (l == null) {
+            firstNode = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+    }
+
+    private static class Node<E> {
```

Comment:
Don't use getters and setters in class Node. Since Node is an inner class, you can access its private fields directly in the outer class.